### PR TITLE
Support full alphabet for user-defined shortcuts

### PR DESCRIPTION
**Pull request by errikos**
POST requests use restricted alphabet (auto-generated shortcuts).
Rest request types (PUT/PATCH) use full alphabet.

Depends on #22

### CODE REVIEW

File: ursh/schemas.py [hunk 1]
```diff
@@ -3,21 +3,33 @@
...
 
 
-class TokenSchema(Schema):
+class SchemaBase(Schema):
+    class Meta:
+        strict = True
+
+    @staticmethod
+    def handle_error(error, data):
+        if type(data) == dict:
```
Comment (ThiefMaster) [inline ursh/schemas.py]: `isinstance(data, dict)` but this type check seems rather strange.. why do we need this?

Comment (errikos) [reply ursh/schemas.py]: Well, the way the handlers are written, `data` is always a `dict`. Maybe left-over from debugging.

File: ursh/schemas.py [hunk 2]
```diff
@@ -41,6 +41,11 @@ class TokenSchema(SchemaBase):
 
 
 class URLSchema(SchemaBase):
+    """Schema class to validate URLs.
+
+    Note: use one of the sub-classes below for validation, depending on the shortcut requirements.
```
Comment (ThiefMaster) [inline ursh/schemas.py]: you could use the full set in here and only use a subclass for restricted where you override the validation method (assuming it doesn't go up the mro and calls the bases of overridden validation methods as well)

Comment (errikos) [reply ursh/schemas.py]: Was fighting with that dilemma. Ended up doing it like I did in order to be more "explicit".

**Pull request merged**